Models/ImageClassification/DenseNet.swift

```diff
@@ -0,0 +1,148 @@
+    public struct Conv: Layer {
+        public var bNorm: BatchNorm<Float>
+        public var conv: Conv2D<Float>
+
+        public init(
+            filterSize: Int,
+            stride: Int = 1,
+            inputFilterCount: Int,
+            outputFilterCount: Int
+        ) {
+            bNorm = BatchNorm(featureCount: inputFilterCount)
+            conv = Conv2D(
+                filterShape: (filterSize, filterSize, inputFilterCount, outputFilterCount),
+                strides: (stride, stride),
+                padding: .same
+            )
+        }
+
+        @differentiable
+        public func callAsFunction(_ input: Tensor<Float>) -> Tensor<Float> {
+            return conv(relu(bNorm(input)))
+        }
+    }
+
+    // Seperate structure for the 1x1 and 3x3 conv layers whose collection
+    // make a dense block
+    public struct ConvPair: Layer {
+        public var conv1x1: Conv
+        public var conv3x3: Conv
+
+        public init(inputFilterCount: Int, growthRate: Int) {
+            conv1x1 = Conv(
+                filterSize: 1,
+                inputFilterCount: inputFilterCount,
+                outputFilterCount: inputFilterCount * 2
+            )
+            conv3x3 = Conv(
+                filterSize: 3,
+                inputFilterCount: inputFilterCount * 2,
+                outputFilterCount: growthRate
+            )
+        }
+
+        @differentiable
+        public func callAsFunction(_ input: Tensor<Float>) -> Tensor<Float> {
+            let conv1Output = conv1x1(input)
+            let conv3Output = conv3x3(conv1Output)
+            return conv3Output.concatenated(with: input, alongAxis: -1)
+        }
+    }
+
+    public struct DenseBlock: Layer {
+        public var pairs: [ConvPair] = []
+
+        public init(repitionCount: Int, growthRate: Int = 32, inputFilterCount: Int) {
+            for i in 0 ..< repitionCount {
+                let filterCount = inputFilterCount + i * growthRate
+                pairs.append(ConvPair(inputFilterCount: filterCount, growthRate: growthRate))
+            }
+        }
+
+        @differentiable
+        public func callAsFunction(_ input: Tensor<Float>) -> Tensor<Float> {
+            let pairsReduced = pairs.differentiableReduce(input) { last, layer in
+                layer(last)
+            }
+            return pairsReduced
+        }
+    }
+
+    public struct TransitionLayer: Layer {
+        public var conv: Conv
+        public var pool: AvgPool2D<Float>
+
+        public init(inputFilterCount: Int) {
+            conv = Conv(
+                filterSize: 1,
+                inputFilterCount: inputFilterCount,
+                outputFilterCount: inputFilterCount / 2
+            )
+            pool = AvgPool2D(poolSize: (2, 2), strides: (2, 2), padding: .same)
+        }
+
+        @differentiable
+        public func callAsFunction(_ input: Tensor<Float>) -> Tensor<Float> {
+            return input.sequenced(through: conv, pool)
```

Comment:
```suggestion
            input.sequenced(through: conv, pool)
```
We can omit the `return` in functions whose body has a single expression.